source/autocomplete/autocomplete_api.cpp

```diff
@@ -315,15 +325,28 @@ void autocomplete(navitia::PbCreator& pb_creator, const std::string &q,
     }
 
 
+    auto main_stop_areas = get_main_stop_area(d);
+
     //Sort the list of objects (sort by object type , score, quality and name)
     //delete unwanted objects at the end of the list
-    auto compare_attributs = [](pbnavitia::PtObject a, pbnavitia::PtObject b)->bool {
+    auto compare_attributs = [&main_stop_areas](pbnavitia::PtObject a, pbnavitia::PtObject b)->bool {
         //Sort by object type
         if (a.embedded_type() != b.embedded_type()){
             const auto a_order = get_embedded_type_order(a.embedded_type());
             const auto b_order = get_embedded_type_order(b.embedded_type());
             return  a_order < b_order;
         }
+        //prioritize main stop_area over the others stop_area
+        if((a.embedded_type() == pbnavitia::STOP_AREA) && (b.embedded_type() == pbnavitia::STOP_AREA)){
+            bool a_is_main_sa = main_stop_areas.find(a.stop_area().uri()) != main_stop_areas.end();
```

Comment:
minor: to simplify readability
```suggestion
            auto a_is_main_sa = main_stop_areas.count(a.stop_area().uri());
```

Reply:
That's slower. I prefer the find personally.

Reply:
That's a set, so not slower, forget my comment, count is better.